battle-service/src/main/java/com/company/battle/utils/GameUtils.java

```diff
@@ -0,0 +1,155 @@
+package com.company.battle.utils;
+
+import com.company.common.models.GameEntity;
+import com.company.common.models.MoveEntity;
+import com.company.common.models.UserEntity;
+import com.company.common.models.enums.GameStatus;
+
+import java.util.Map;
+import java.util.UUID;
+
+public class GameUtils {
```

Comment:
try to split this class to multiple utils/services.

GameValidationService
GameWinnerService